### PR TITLE
[Semaphore] Add documentation about semaphore configuration

### LINKED ISSUE #16707: [FrameworkBundle] Add semaphore configuration
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#44155
| PR author(s) | @jderusse
| Merged in    | 6.1

### PR DESCRIPTION
**Pull request by jderusse**
fix #16707

Deeply inspired from lock's configuration

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -3082,6 +3082,81 @@ name
...
+
+.. versionadded:: 6.1
+
+    The ``semaphore`` option was introduced in Symfony 6.1.
+
+**type**: ``string`` | ``array``
+
+The default semaphore adapter. Store's DSN are also allowed.
+
+.. _reference-lock-enabled:
```
Comment (wouterj) [inline reference/configuration/framework.rst]:
```suggestion
.. _reference-semaphore-enabled:
```

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -3082,6 +3082,81 @@ name
...
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``true``
+
+Whether to enable the support for semaphore or not. This setting is
+automatically set to ``true`` when one of the child settings is configured.
+
+.. _reference-lock-resources:
```
Comment (wouterj) [inline reference/configuration/framework.rst]:
```suggestion
.. _reference-semaphore-resources:
```

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -3082,6 +3082,81 @@ name
...
+
+resources
+.........
+
+**type**: ``array``
+
+A list of semaphore stores to be created by the framework extension.
```
Comment (wouterj) [inline reference/configuration/framework.rst]:
```suggestion
A map of semaphore stores to be created by the framework extension, with
the name as key and DSN as value:
```

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -3082,6 +3082,81 @@ name
...
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/semaphore.yaml
+        framework:
+            semaphore: '%env(SEMAPHORE_DSN)%'
```
Comment (wouterj) [inline reference/configuration/framework.rst:3123]:
```suggestion
            semaphore:
                resources:
                    default: '%env(SEMAPHORE_DSN)%'
```

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -3082,6 +3082,81 @@ name
...
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:semaphore>
+                    <framework:resource>%env(SEMAPHORE_DSN)%</framework:resource>
```
Comment (wouterj) [inline reference/configuration/framework.rst]:
```suggestion
                    <framework:resource name="default">%env(SEMAPHORE_DSN)%</framework:resource>
```

**Review by wouterj — APPROVED** [review]
Thanks for the fast response!

I've left some minor suggestions (to fix a copypaste issue and show a full example)

File: reference/configuration/framework.rst (hunk 4 above)
Comment (jderusse) [reply reference/configuration/framework.rst:3123]: Both are correct, but the first one (`semaphore: '%env(SEMAPHORE_DSN)%'`) is easiest to read IMHO and consistent with `lock` documentation

**Comment by javiereguiluz** [comment]
Jérémy, thanks for this nice contribution!

**Pull request merged**